devolutions-gateway/src/recording/muxer.rs

```diff
@@ -0,0 +1,43 @@
+use std::{
+    path::{Path, PathBuf},
+    sync::Arc,
+    time::Duration,
+};
+
+use anyhow::Context;
+
+pub async fn remux(input: impl AsRef<Path>) -> anyhow::Result<()> {
+    let path = Arc::new(input.as_ref().to_owned());
+    // Could potentially be a very slow operation
+    tokio::task::spawn_blocking(|| remux_impl(path)).await??;
+    Ok(())
+}
+
+fn remux_impl(input: Arc<PathBuf>) -> anyhow::Result<()> {
+    std::thread::sleep(Duration::from_secs(1));
+    let input_to_remux = input.as_ref();
+    let output = input_to_remux
+        .as_path()
+        .ancestors()
+        .nth(1)
+        .with_context(|| "failed to create muxer cache file")?
+        .join(format!(
+            "muxer_cache_{}",
+            input_to_remux
+                .file_name()
+                .ok_or(anyhow::anyhow!("input file has no file name"))?
+                .to_string_lossy()
+        ));
+    cadeau::xmf::muxer::webm_remux(input_to_remux, output.as_path()).with_context(|| {
+        format!(
+            "failed to remux file: {}, output: {}",
+            input_to_remux.display(),
+            output.display(),
+        )
+    })?;
+
+    std::fs::copy(output.as_path(), input_to_remux).with_context(|| "failed to copy remuxed file")?;
+    std::fs::remove_file(output).with_context(|| "failed to remove temporary file")?;
+    debug!(?input_to_remux, "Successfully Remuxed file");
+    Ok(())
```

Comment:
suggestion: You can simplify this code a lot by using `Utf8PathBuf`, the `parent` method, `std::fs::rename` a a few intermediate variables. Something like this:

```rust
pub async fn remux(input_path: Utf8PathBuf) -> anyhow::Result<()> {
    use std::time::Duration;

    // We need to sleep because […]
    tokio::time::sleep(Duration::from_secs(1)).await;

    // CPU-intensive operation potentially lasting much more than 100ms.
    tokio::task::spawn_blocking(move || remux_impl(input_path)).await??;

    return Ok(());

    fn remux_impl(input_path: Utf8PathBuf) -> anyhow::Result<()> {
        let input_file_name = input_path
            .file_name()
            .ok_or(anyhow::anyhow!("input file has no file name"))?;

        let remuxed_file_name = format!("remuxed_{input_file_name}");

        let output_path = input_path
            .parent()
            .context("failed to retrieve parent folder")?
            .join(remuxed_file_name);

        cadeau::xmf::muxer::webm_remux(&input_path, &output_path)
            .with_context(|| format!("failed to remux file {input_path} to {output_path}"))?;

        std::fs::rename(&output_path, &input_path).context("failed to override remuxed file")?;

        debug!(%input_path, "Successfully remuxed video recording");

        Ok(())
    }
}
```

Reply:
updated